### ISSUE TITLE
Move MPI Parallelisation to ParGrid and remove RectGrid

### ISSUE THREAD

**Issue by MarionBWeinzierl** [nextsimhub/nextsimdg#448]
Following the discussion in #442 , we need to move the MPI parallelisation done in #331 over to the parametric grid, and then remove the rectangular grid.

This should be done after #331 and #432 were merged into `develop`.

**Comment by MarionBWeinzierl** [comment]
Information about the grid from @timspainNERSC : https://github.com/nextsimhub/nextsimdg/blob/paragrid_doc/docs/model_grids.rst 

Positions of variables on the grid:
![grid](https://github.com/nextsimhub/nextsimdg/assets/56403724/30908438-5f43-4028-9ef3-6e6529e77d62)

**Comment by TomMelt** [comment]
Paragrid has been parallelized in #560
